src/lib.rs

```diff
@@ -1,350 +1,434 @@
 use std::collections::HashMap;
+use std::collections::HashSet;
+use std::hash::Hash;
 
-/// Compute the gini impurity of a dataset. 
-/// 
+/// Compute the gini impurity of a dataset.
+///
 /// Returns a float, 0 representing a perfectly pure dataset. Normal distribution: ~0.33
-/// 
+///
 /// By default, any empty dataset will return a gini of 1.0. This may be unexpected behaviour.
 /// ```
 /// use parsnip::gini;
-/// assert_eq!(gini(&vec![0, 0, 0, 1]), 0.375);
+/// assert_eq!(gini(&vec![0_usize, 0, 0, 1]), 0.375);
```

Comment:
This should work on `i32` now, right? I think we can omit the `_usize` suffix in doctests to keep them simple.

Reply:
Yup, cleaned up!